dissect/target/plugins/filesystem/ntfs/mft.py

```diff
@@ -167,55 +176,64 @@ def noaggr(records: list[Record]) -> Iterator[Record]:
         elif macb:
             aggr = macb_aggr
 
-        for fs in self.target.filesystems:
-            if fs.__type__ != "ntfs":
-                continue
-
-            # If this filesystem is a "fake" NTFS filesystem, used to enhance a
-            # VirtualFilesystem, The driveletter (more accurate mount point)
-            # returned will be that of the VirtualFilesystem. This makes sure
-            # the paths returned in the records are actually reachable.
-            drive_letter = get_drive_letter(self.target, fs)
-            volume_uuid = get_volume_identifier(fs)
-
+        if fs is not None:
             try:
-                for record in fs.ntfs.mft.segments():
-                    try:
-                        inuse = bool(record.header.Flags & FILE_RECORD_SEGMENT_IN_USE)
-                        owner, _ = get_owner_and_group(record, fs)
-                        resident = False
-                        size = None
-
-                        if not record.is_dir():
-                            for data_attribute in record.attributes.DATA:
-                                if data_attribute.name == "":
-                                    resident = data_attribute.resident
-                                    break
-
-                            size = get_record_size(record)
-
-                        for path in record.full_paths():
-                            path = f"{drive_letter}{path}"
-                            yield from aggr(
-                                self.mft_records(
-                                    drive_letter=drive_letter,
-                                    record=record,
-                                    segment=record.segment,
-                                    path=path,
-                                    owner=owner,
-                                    size=size,
-                                    resident=resident,
-                                    inuse=inuse,
-                                    volume_uuid=volume_uuid,
-                                    record_formatter=record_formatter,
-                                )
+                filesystem = self.ntfs_filesystems[fs]
+            except KeyError:
+                self.target.log.error("NTFS filesystem with index number %s does not exist", fs)
+                return
+
+            yield from self.segments(filesystem, record_formatter, aggr, start, end)
+        else:
+            for filesystem in self.ntfs_filesystems.values():
+                yield from self.segments(filesystem, record_formatter, aggr, start, end)
+
+    def segments(self, fs: NtfsFilesystem, record_formatter: Callable, aggr: Callable, start: int, end: int):
```

Comment:
Missing return type hint

Reply:
Should that also be added to the other functions that miss return type hints `-> Iterator[Record]:`?

* https://github.com/Zawadidone/dissect.target/blob/feature/mft_segments/dissect/target/plugins/filesystem/ntfs/mft.py#L318
* https://github.com/Zawadidone/dissect.target/blob/feature/mft_segments/dissect/target/plugins/filesystem/ntfs/mft.py#L307
* https://github.com/Zawadidone/dissect.target/blob/feature/mft_segments/dissect/target/plugins/filesystem/ntfs/mft.py#L238

Reply:
Didn't know there were so many 😄 yes please.